content/response_integrations/power_ups/tools/actions/CreateSiemplifyTask.py

```diff
@@ -15,91 +15,139 @@
-            "name": task_content,
-            "dueDate": "",
-            "dueDateUnixTimeMs": client_time,
-            "caseId": siemplify.case_id,
-        }
-        task_url = X5_TASK_URL
-
-    add_task = siemplify.session.post(
-        task_url.format(siemplify.API_ROOT),
-        json=json_payload,
-    )
-    add_task.raise_for_status()
-
-    output_message = f"A new task has been created for the user: {assign_to}.\nThis task should be handled in the next {duration} mintues"
-    siemplify.end(output_message, True)
+class CreateSiemplifyTaskAction(Action):
+    def __init__(self):
+        super().__init__(ACTION_NAME)
+
+    def _extract_action_parameters(self):
+        self.params.assign_to = extract_action_param(
+            self.soar_action,
+            param_name="Assign To",
+            is_mandatory=True,
+            print_value=True,
+        )
+        self.params.task_content = extract_action_param(
+            self.soar_action,
+            param_name="Task Content",
+            is_mandatory=True,
+            print_value=True,
+        )
+        self.params.duration = extract_action_param(
+            self.soar_action,
+            param_name="SLA (in minutes)",
+            is_mandatory=False,
+            print_value=True,
+        )
+        self.params.task_title = extract_action_param(
+            self.soar_action,
+            param_name="Task Title",
+            is_mandatory=False,
+            print_value=True,
+        )
+        self.params.due_date = extract_action_param(
+            self.soar_action,
+            param_name="Due Date",
+            is_mandatory=False,
+            print_value=True,
+        )
+
+    def _validate_params(self):
+        if not self.params.due_date and not self.params.duration:
+            raise ValueError(
+                'either "Due Date" or "SLA (in minutes)" parameter should have a value.'
+            )
+
+    def _init_api_clients(self) -> None:
+        """Initialize API clients if required (placeholder)."""
+
+    def _perform_action(self, _) -> None:
+        client_time = self._client_time()
+
+        self._create_task(client_time)
+
+        due_date_info = (
+            f"by {self.params.due_date}"
+            if self.params.due_date
+            else f"in the next {self.params.duration} minutes"
+        )
+        self.output_message = (
+            f"A new task has been created for the user: {self.params.assign_to}.\n"
+            f"This task should be handled {due_date_info}"
+        )
+        self.result_value = True
+
+    def _client_time(self):
+        client_time = 0
+
+        if self.params.due_date:
+            dt_due_date = parse(self.params.due_date)
+            client_time = int(dt_due_date.timestamp() * 1000)
+        elif self.params.duration:
+            time_now_epoch = int(time.time() * 1000)
+            client_time = time_now_epoch + (int(self.params.duration) * 1000 * 60)
+        return client_time
```

Comment:
duration param is guaranteed here due to validation so we can drop elif and return early.
```suggestion
    def _compute_due_time(self) -> int:
        """Return due time in epoch millis based on given parameters."""
        if self.params.due_date:
            return int(parse(self.params.due_date).timestamp() * 1000)

        minutes = int(self.params.duration)
        return int(time.time() * 1000) + minutes * 60 * 1000
```